jl4-lsp/src/LSP/L4/Viz/VizExpr.hs

```diff
@@ -160,3 +150,13 @@ I haven't actually tried setting up such a automated test though.
 
 If that doesn't end up panning out, I will switch back to aeson.
 -}
+
+-------------------------------------------------------------
+-- Utils
+-------------------------------------------------------------
+{- | Adapted from
+https://github.com/NorfairKing/autodocodec/blob/e939442995debec6d0e014bfcc45449b3a2cb6e6/autodocodec-api-usage/src/Autodocodec/Usage.hs#L827
+-}
```

Comment:
I don't think you need to add this link here, but it's fine if you do.

Reply:
btw, idk if I had already asked this but have you checked the license of that code? norfairking sometimes has weird one off licenses that trap commercial users.